migrate.py

```diff
@@ -649,6 +690,314 @@ def copy_tests(plugin, coll, spec, args):
+                    plugin_data = read_text_from_file(full_path)
+                    plugin_data_new = plugin_data[:]
+
+                    plugin_data_new, import_dependencies = rewrite_imports(plugin_data_new, collection, spec, namespace)
+                    plugin_data_new, docs_dependencies = rewrite_doc_fragments(plugin_data_new, collection, spec, args)
+
+                    for dep in docs_dependencies + import_dependencies:
+                        integration_tests_add_to_deps(collection, dep)
+
+                    write_text_into_file(dest, plugin_data_new)
+                elif ext in ('.ps1',):
+                    # FIXME
+                    pass
+                elif ext in ('.yml', '.yaml'):
+                    rewrite_yaml(full_path, dest, namespace, collection, spec)
+                elif ext in ('.sh',):
+                    rewrite_sh(full_path, dest, namespace, collection, spec)
+                elif ext in ('.cfg',):
+                    rewrite_ini(full_path, dest, namespace, collection, spec)
+                else:
+                    shutil.copy2(full_path, dest)
+
+
+def rewrite_sh(full_path, dest, namespace, collection, spec):
+    sh_key_map = {
+        'ANSIBLE_CACHE_PLUGIN': 'cache',
+        'ANSIBLE_CALLBACK_WHITELIST': 'callback',
+        'ANSIBLE_INVENTORY_CACHE_PLUGIN': 'cache',
+        'ANSIBLE_STDOUT_CALLBACK': 'callback',
+        'ANSIBLE_STRATEGY': 'strategy',
+        '--become-method': 'become',
+        '-c': 'connection',
+        '--connection': 'connection',
+    }
+
+    contents = read_text_from_file(full_path)
+    for key, plugin_type in sh_key_map.items():
+        if not contents.find(key):
+            continue
+        for coll in spec.keys():
+            plugins = get_plugins_from_collection(coll, plugin_type, spec)
+            for plugin_name in plugins:
+                if not contents.find(plugin_name):
+                    continue
+                # FIXME list
+                new_plugin_name = get_plugin_fqcn(namespace, coll, plugin_name)
+                contents = contents.replace(key + '=' + plugin_name, key + '=' + new_plugin_name)
+                contents = contents.replace(key + ' ' + plugin_name, key + ' ' + new_plugin_name)
+                if collection != coll:
+                    integration_tests_add_to_deps(collection, coll)
+
+    write_text_into_file(dest, contents)
+    shutil.copystat(full_path, dest)
+
+
+def rewrite_ini(src, dest, namespace, collection, spec):
+    ini_key_map = {
+        'defaults': {
+            'callback_whitelist': 'callback',
+            'fact_caching': 'cache',
+            'strategy': 'strategy',
+            'stdout_callback': 'callback',
+        },
+        'inventory': {
+            'cache_plugin': 'cache',
+            'enable_plugins': 'inventory',
+        }
+    }
+
+    config = configparser.ConfigParser()
+    config.read(src)
+    for section in config.sections():
+        try:
```

Comment:
Since you just ignore the exception, you can use suppress here:
```suggestion
        with contextlib.suppress(KeyError):
```